utils/azure/create-scale-set.sh

```diff
@@ -0,0 +1,83 @@
+#!/usr/bin/env bash
+
+# Shamelessly stolen from https://github.com/lsds/sgx-lkl
+
+# This script creates one of two VM scale set variants for use in Azure Pipelines.
+# After the scale set is created, a corresponding pool has to be created
+# in the Azure DevOps project that uses the scale set.
+# The two supported variants are "linux" and "windows"
+# The created resource group has the name '<vmss-name>-rg' where the VMSS name
+# must be specified as script argument.
+
+# Notes on OS and temporary disk:
+# The OS disk can either be ephemeral or persisted.
+# Ephemeral OS disks are backed by local non-cloud storage and can
+# at most be the size of the cached storage (e.g. DC4s_v2 has 86 GiB cache size).
+# See also https://docs.microsoft.com/en-us/azure/virtual-machines/linux/ephemeral-os-disks.
+# Persisted OS disks can have arbitrary sizes but have higher latency
+# and make reset/reimaging operations of scale sets slower.
+# The temporary disk is always backed by local non-cloud storage and has
+# similar characteristics as an ephemeral OS disk. The size of the disk
+# is prescribed by the VM sku.
+
+# Notes on Azure Pipelines and its scale set agent feature:
+# The Azure Pipelines agent (including the job folder and home directory)
+# is running from the OS disk when using the scale set agent feature.
+# It is currently not possible to use the (often larger) temporary disk.
+# Ephemeral OS disks are currently not supported for DC v2 series VMs.
+# This means the best option currently is to use persisted OS disks for
+# the SGX scale set and ephemeral OS disks for the non-SGX scale set.
+# Once ephemeral OS disks are supported for DC v2 series VMs this should
+# be re-evaluated, keeping in mind that the maximum ephemeral OS disk size
+# for DC4s_v2 would be 86 GiB.
+
+set -e
+
+expected_args=4
+if [[ $# != $expected_args ]]; then
+    echo "Usage: $0 <subscription> <region> <vmss-name> linux|windows"
+    echo "Example: $0 7150ce20-6afe... eastus llvm-build linux"
```

Comment:
fwiw you can usually give sub names rather than sub ids to azure CLI, they tend to be easier to remember.

Reply:
nice

Reply:
Copy&paste from the LKL one. :)
But thanks for the tip.